spec/models/user_spec.rb

```diff
@@ -190,18 +339,143 @@
+      it 'returns true' do
+        allow(subject).to receive(:is_admin).and_return(false)
+        allow(subject).to receive(:is_manager).and_return(false)
+        expect(item).not_to receive(:can_manage_content?)
+        manager_roles = ["dor-administrator", "sdr-administrator"]
+        manager_roles.each do |role|
+          allow(subject).to receive(:roles).and_return([role])
+          allow(item).to receive(:can_manage_item?).with([role]).and_return(true)
+          # can_manage_item? supercedes can_manage_content?
+          expect(subject.can_manage_object?(item)).to be true
+        end
+      end
+    end
+    context 'when user has no authorized role' do
+      it 'returns false' do
+        allow(subject).to receive(:is_admin).and_return(false)
+        allow(subject).to receive(:is_manager).and_return(false)
+        viewer_roles = ["dor-viewer", "sdr-viewer"]
+        viewer_roles.each do |role|
+          allow(subject).to receive(:roles).and_return([role])
+          allow(item).to receive(:can_manage_item?).with([role]).and_return(false)
+          allow(item).to receive(:can_manage_content?).with([role]).and_return(false)
+          expect(subject.can_manage_object?(item)).to be false
+        end
+      end
+    end
+  end
+
+  describe '#can_view_object?' do
+    let(:druid) { 'druid:hv992ry2431' }
+    let(:item) do
+      item = instantiate_fixture(druid, Dor::AdminPolicyObject)
+      # allow(Dor).to receive(:find).with(@druid).and_return(@item)
+      # allow(Dor::Item).to receive(:find).with(@druid).and_return(@item)
+      item
+    end
+    it 'returns false' do
+      expect(subject.can_view_object?(item)).to be false
+    end
+    context 'when admin' do
+      it 'returns true' do
+        expect(subject).to receive(:is_admin).and_return(true)
+        expect(subject).not_to receive(:is_manager)
+        expect(subject).not_to receive(:is_viewer)
+        expect(subject.can_view_object?(item)).to be true
+      end
+    end
+    context 'when manager' do
+      it 'returns true' do
+        expect(subject).to receive(:is_admin).and_return(false)
+        expect(subject).to receive(:is_manager).and_return(true)
+        expect(subject).not_to receive(:is_viewer)
+        expect(subject.can_view_object?(item)).to be true
+      end
+    end
+    context 'when viewer' do
+      it 'returns true' do
+        expect(subject).to receive(:is_admin).and_return(false)
+        expect(subject).to receive(:is_manager).and_return(false)
+        expect(subject).to receive(:is_viewer).and_return(true)
+        expect(subject.can_view_object?(item)).to be true
+      end
+    end
+    context 'when user has an authorized role' do
+      it 'returns true' do
+        allow(subject).to receive(:is_admin).and_return(false)
+        allow(subject).to receive(:is_manager).and_return(false)
+        allow(subject).to receive(:is_viewer).and_return(false)
+        viewer_roles = ["dor-viewer", "sdr-viewer"]
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.